apps/api_banking/lib/api_banking/users/create.ex

```diff
@@ -0,0 +1,74 @@
+defmodule ApiBanking.Users.Create do
+  alias ApiBanking.Repo
+  alias Ecto.Multi
+  alias ApiBanking.Users.Schemas.User
+  alias ApiBanking.Accounts.Schemas.Account
+
+  require Logger
+
+  def create_user(struct) when is_struct(struct) do
+    params = %{email: struct.email, name: struct.name, password_hash: struct.password_hash}
+    create_user(params)
+  end
+
+  def create_user(params) do
+    Logger.info("Inserting new User")
+
+    with %{valid?: true} = changeset <- User.changeset(params),
+         {:ok, user} <- create_user_and_account(changeset) do
+      {:ok, user}
+    else
+      %{valid?: false} = changeset ->
+        msg_error =
+          Ecto.Changeset.traverse_errors(changeset, fn {msg, opts} ->
+            Enum.reduce(opts, msg, fn {key, value}, acc ->
+              String.replace(acc, "%{#{key}}", to_string(value))
+            end)
+          end)
+
+        {:error, %{msg_error: msg_error}}
+    end
+  rescue
+    Ecto.ConstraintError ->
+      Logger.error("Email already in use")
+
+      {:error, :email_conflict}
+  end
+
+  defp create_user_and_account(user_changeset) do
+    Multi.new()
+    |> Multi.insert(:create_user, user_changeset)
+    |> Multi.run(:create_account, fn repo, %{create_user: user} ->
+      insert_account(repo, user)
+    end)
+    |> Multi.run(:preload_data, fn repo, %{create_user: user} ->
+      preload_data(repo, user)
+    end)
+    |> run_transaction()
+  end
+
+  defp insert_account(repo, user) do
+    code = generate_code()
+
+    Account.changeset(%{user_id: user.id, account_code: code, balance: 100_000})
```

Comment:
O credo (https://github.com/rrrene/credo) é uma ferramenta que a gente usa pra análise estática do código e força um estilo específico de se escrever o Elixir pra melhorar a legibilidade. Apesar de não estar errado, ele reclama de pipes que começam com funções, a princípio, todo pipe deveria começar com um valor puro quando possível, nesse caso:
```elixir
%{user_id: user.id, account_code: code, balance: 100_000}
|> Account.changeset()
|> repo.insert()
```

Ele também reclama de pipes que só tem um pipe, ex:
```elixir
changeset
|> Repo.insert()

# Deveria ser
Repo.insert(changeset)
```

O credo, o formatter e o dialyzer são as ferramentas de análise estática que fazem parte do nosso CI, se um deles não passar o código não vai pra master. Pode valer a pena tu instalar o credo aqui como dependencia pra ver as sugestões dele.